pvp-faq.md

```diff
@@ -29,9 +29,13 @@ The most apparent somewhat superficial difference between SemVer and the
 PVP is which components of the version number are interpreted as
 major, minor, and patch-level numerals:
 
- - PVP: *MAJOR.MAJOR.MINOR.PATCH*
+ - PVP: *MAJOR.MAJOR.MINOR.PATCH* [^1]
  - SemVer: *MAJOR.MINOR.PATCH*
 
+[^1]: In PVP, components beyond the third are optional, to be used in any
+      way the package maintainer sees fit. In practice, four-component
+      versioning (`A.B.C.D`) is standard.
+
```

Comment:
Is this true? My recollection is hackage and/or cabal enforce we have no more than four components.

Reply:
That's irrelevant to the spec what cabal/hackage enforce.

Reply:
More generally, what I wanted to convey is “the standard says A.B.C, in real life you will 99% find A.B.C.D”. If it is not clear, I welcome rewording suggestions.

Reply:
https://hackage.haskell.org/package/regex-pcre-builtin-0.95.2.3.8.44 6 components

Reply:
Hate to be a language lawyer, but I find it confusing for the opposite direction. To me, the line:

> In PVP, components beyond the third are optional

implies that at least three components are required. But PVP says:

> A package version number SHOULD have the form A.B.C, and MAY optionally have any number of additional components

where **SHOULD** means [recommended](https://datatracker.ietf.org/doc/html/rfc2119). In practice, quite a few packages only use as many components "as-needed" e.g. [hlint](https://hackage.haskell.org/package/hlint), [optics](https://hackage.haskell.org/package/optics). In fact, I'm not sure PVP excludes using a single component, and just bumping it for every change.